### PR TITLE
Add basic content conformance CLI

### PR DESCRIPTION
**Pull request by nayres**
🎟️ [Asana Task](https://app.asana.com/0/1203331966445516/1203331966445545)

---

## Description

Build out the CLI wrapper in `cli.ts`. Should handle logging to stdout and passing CLI options along to the ContentConformanceRunner.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/content-conformance/tsconfig.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "moduleResolution": "NodeNext",
     "jsx": "react",
     "outDir": "./dist",
-    "declaration": true
+    "declaration": true,
+    "esModuleInterop": true,
```
Comment (nayres) [inline packages/content-conformance/tsconfig.json:10]: These trailing commas help with reducing noise in PRs, and providing proper blame in git.

File: packages/content-conformance/src/cli.ts [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+        console.log(chalk.redBright(error.stack))
+      })
 
-  console.log(runner.config)
-}
-
-main()
-
-export {}
+    console.log(chalk.cyanBright(`Config: ${runner.config}`))
```
Comment (nayres) [inline packages/content-conformance/src/cli.ts:51]: As of now these all of these logs are mostly just placeholders.

File: packages/content-conformance/package.json [hunk 3]
```diff
@@ -16,9 +16,14 @@
     "find-up": "^6.3.0",
     "globby": "^13.1.2",
     "remark-mdx": "^1.6.22",
+    "ts-node": "^10.9.1",
```
Comment (BRKalow) [inline packages/content-conformance/package.json:19]: Was this for testing purposes? If so, we might be able to use our `hc-tools` CLI (a light wrapper around `ts-node`) to run this without building.

File: packages/content-conformance/src/cli.ts [hunk 4]
```diff
@@ -1,17 +1,53 @@
 #!/usr/bin/env node
+import yargs from 'yargs'
+import { hideBin } from 'yargs/helpers'
```
Comment (BRKalow) [inline packages/content-conformance/src/cli.ts:3]: nice, TIL about `hideBin`!

File: packages/content-conformance/src/cli.ts [hunk 5]
```diff
@@ -1,17 +1,53 @@
...
+        console.log(
+          chalk.bold.greenBright(`Running content conformance checks on:`)
+        )
+        argv.files.forEach((file: string) => {
+          console.log(chalk.green(`- ${file}`))
+        })
+      })
+      .catch((error) => {
+        console.log(chalk.redBright(error.stack))
+      })
```
Comment (BRKalow) [inline packages/content-conformance/src/cli.ts:49]: I think it'd be ideal to stick with one syntax for handling async code here:

```suggestion
		try {
			await runner.init()

			console.log(
				chalk.bold.greenBright(`Running content conformance checks on:`)
			)
			argv.files.forEach((file: string) => {
				console.log(chalk.green(`- ${file}`))
			})
		} catch (error) {
			console.log(chalk.redBright(error.stack))
		}
```

File: packages/content-conformance/src/cli.ts [hunk 6]
```diff
@@ -1,17 +1,53 @@
...
+    await runner
+      .init()
+      .then(() => {
+        console.log(
+          chalk.bold.greenBright(`Running content conformance checks on:`)
+        )
+        argv.files.forEach((file: string) => {
+          console.log(chalk.green(`- ${file}`))
+        })
+      })
```
Comment (BRKalow) [inline packages/content-conformance/src/cli.ts:46]: Just calling out that `files` is optional, but it looks like we have a default of `[]` so this should be okay 👍

File: packages/content-conformance/tsconfig.json (hunk 1 above)
Comment (BRKalow) [inline packages/content-conformance/tsconfig.json:10]: Was `esModuleInterop` necessary for one of the new dependencies?

File: packages/content-conformance/src/cli.ts [hunk 7]
```diff
@@ -1,17 +1,53 @@
...
+      default: undefined,
+      type: 'string',
+    },
+    config: {
+      description: 'Path to config file',
+      default: undefined,
+      type: 'string',
+    },
+    files: {
+      description: 'Path to config file',
```
Comment (BRKalow) [inline packages/content-conformance/src/cli.ts:25]: Looks like we'll want to update this description 👀

**Review by BRKalow — COMMENTED** [review]
Great start! What do you think will be the best way to test this? 💭

File: packages/content-conformance/tsconfig.json (hunk 1 above)
Comment (nayres) [reply packages/content-conformance/tsconfig.json:10]: No it wasn't. It's an artifact of me trying to get rid of TS errors :D

File: packages/content-conformance/src/cli.ts (hunk 7 above)
Comment (nayres) [reply packages/content-conformance/src/cli.ts:25]: Updated!

File: packages/content-conformance/package.json (hunk 3 above)
Comment (nayres) [reply packages/content-conformance/package.json:19]: Removed!

File: packages/content-conformance/package.json [hunk 8]
```diff
@@ -19,7 +19,10 @@
     "unist-util-visit": "^4.1.1",
     "vfile": "^5.3.5",
     "vfile-reporter": "^7.0.4",
-    "yaml": "^2.1.3",
```
Comment (BRKalow) [inline packages/content-conformance/package.json:22]: We still want this dependency, it looks like this might have been a merge conflict casualty.

Comment (nayres) [reply packages/content-conformance/package.json:22]: Added it back!

File: packages/content-conformance/src/cli.ts [hunk 9]
```diff
@@ -1,17 +1,66 @@
...
 /**
  * The CLI interface for our content conformance runner. Should remain a pass-through to ContentConformanceRunner if at all possible to keep the CLI and JS API in-sync
  */
-async function main() {
-  const runner = new ContentConformanceRunner()
 
-  await runner.init()
+type ErrorWithMessage = {
+  message: string
+}
```
[the remark this answers is not in the document]
Comment (nayres) [reply packages/content-conformance/src/cli.ts]: Removed!

**Pull request merged**